stanford_profile_helper.module

```diff
@@ -609,11 +609,54 @@ function stanford_profile_helper_preprocess_pattern_localfooter(&$variables) {
  * Implements hook_form_FORM_ID_alter().
  */
 function stanford_profile_helper_form_media_library_add_form_embeddable_alter(array &$form, FormStateInterface $form_state) {
+
+  $source_field = $form_state->get('source_field');
+  $embed_code_field = $form_state->get('unstructured_field_name');
   $user = \Drupal::currentUser();
   $authorized = $user->hasPermission('create field_media_embeddable_code')
     || $user->hasPermission('edit field_media_embeddable_code');
 
-  if (isset($form['container']['field_media_embeddable_code'])) {
-    $form['container']['field_media_embeddable_code']['#access'] = $authorized;
+  if (isset($form['container'][$embed_code_field])) {
+    $form['container'][$embed_code_field]['#access'] = $authorized;
+  }
+
+  if (isset($form['container'][$source_field])) {
+    if (!$authorized) {
+      $new_desc = 'Allowed providers: @providers. For custom embeds, please <a href="@snow_form">request support.</a>';
+      $args = $form['container'][$source_field]['#description']->getArguments();
+      $args['@snow_form'] = 'https://stanford.service-now.com/it_services?id=sc_cat_item&sys_id=83daed294f4143009a9a97411310c70a';
+      $form['container'][$source_field]['#description'] = t($new_desc, $args);
+    }
+    $form['container'][$source_field]['#title'] = t('oEmbed URL');
+  }
+
+}
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function stanford_profile_helper_form_media_embeddable_add_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  $user = \Drupal::currentUser();
+  $authorized = $user->hasPermission('create field_media_embeddable_code')
+    || $user->hasPermission('edit field_media_embeddable_code');
+  if (!$authorized) {
+    $source_config = $form_state
+      ->getFormObject()
+      ->getEntity()
+      ->getSource()
+      ->getConfiguration();
+    $source_field = $source_config['oembed_field_name'];
+    $new_desc = 'Allowed providers: @providers. For custom embeds, please <a href="@snow_form">request support.</a>';
+    $args = $form[$source_field]['widget'][0]['value']['#description']->getArguments();
```

Comment:
is this correct? you're getting the `oembed_field_name` field from the config. don't you want to edit the `source_field` instead?

Reply:
Both of those values are identical in the config. I chose the oembed_field_name for readability in this context.

Reply:
we should keep using the `source_field`. the `oembed_field_name` has been removed from the form and should be removed from the config at some point. https://github.com/SU-SWS/stanford_media/blob/8.x-2.x/src/Plugin/media/Source/Embeddable.php#L70

Reply:
Fixed.